src/message.rs

```diff
@@ -28,15 +32,129 @@ fn _message_prettify(message: CString, comment_char: Option<u8>) -> Result<Strin
     Ok(ret.as_str().unwrap().to_string())
 }
 
+/// Collection of trailer key–value pairs.
+///
+/// Use `iter()` to get access to the values.
+pub struct MessageTrailers {
+    raw: raw::git_message_trailer_array,
+    _marker: marker::PhantomData<c_char>,
+}
+
+impl<'pair> MessageTrailers {
```

Comment:
Since the type doesn't have a lifetime parameter any more this shoudl move down to the `iter` method itself, but even on the iter method it can be elided and it can return `MessageTrailersIterator<'_>`